src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = state;
```

Comment:
The current implementation modifies the original 'state' object when 'currentState' is assigned to 'state' directly. To prevent this, you should create a copy of the initial state for 'currentState'.